Automattic/newspack-blocks
# chore(donate): place modal checkout behind RAS flag

### All Submissions:






### Changes proposed in this Pull Request:

Feature rollout of the modal checkout for RAS-enabled sites. Editing the donation billing fields through the Reader Revenue wizard will remain available for every site.

Example of non-modal standard Woo checkout with custom selection of billing fields:

<img width="1279" alt="image" src="https://user-images.githubusercontent.com/820752/221968337-84b42b53-db46-4cf6-99b7-02288890d97b.png">

### How to test the changes in this Pull Request:

1. Make sure your site has the RAS flag: `define( 'NEWSPACK_EXPERIMENTAL_READER_ACTIVATION', true );`
2. Set your RR platform to "Newspack"
3. Donate through the Donate block and confirm the modal checkout works as expected
4. Remove or comment-out the flag
5. Repeat step 3 and confirm you're redirect to the standard Woo checkout
6. Visit the Reader Revenue wizard and tweak the billing fields
7. Confirm the standard Woo checkout is updated with the selected fields

### Other information:
